packages/driver/src/burn-in/index.ts

```diff
@@ -0,0 +1,202 @@
+  // meaning retry logic does NOT need to be applied.
+  if (failedAttemptsCount > 0) {
+    result.nextInitialStrategy = 'RETRY'
+    const remainingAttempts = maxAttempts - totalAttemptsAlreadyExecuted
+
+    // Below variables are used for when strategy is "detect-flake-and-pass-on-threshold" or no strategy is defined
+    let passesRequired = retriesConfig?.strategy === 'detect-flake-and-pass-on-threshold' ?
+      (retriesConfig.passesRequired || 1) :
+      null
+
+    const neededPassingAttemptsLeft = retriesConfig?.strategy === 'detect-flake-and-pass-on-threshold' ?
+      (passesRequired as number) - passedAttemptsCount :
+      null
+
+    // Below variables are used for when strategy is only "detect-flake-but-always-fail"
+    let stopIfAnyPassed = retriesConfig?.strategy === 'detect-flake-but-always-fail' ?
+      (retriesConfig.stopIfAnyPassed || false) :
+      null
+
+    switch (retriesConfig?.strategy) {
+      case 'detect-flake-and-pass-on-threshold':
+        if (passedAttemptsCount >= (passesRequired as number)) {
+          // we met the threshold, so we can stop retrying and pass the test
+          result.outerTestStatus = 'passed'
+          result.reasonToStop = 'PASSED_MET_THRESHOLD'
+        } else if (remainingAttempts < (neededPassingAttemptsLeft as number)) {
+          // we don't have enough remaining attempts to meet the threshold, so we should stop retrying and fail the test
+          result.outerTestStatus = 'failed'
+          result.forceState = currentAttemptResult === 'passed' ? currentAttemptResult : undefined
+          result.reasonToStop = 'FAILED_DID_NOT_MEET_THRESHOLD'
+        } else {
+          // we haven't met the threshold, but we have enough remaining attempts to meet the threshold, so we should retry the test
+          result.final = false
+        }
+
+        break
+      case 'detect-flake-but-always-fail':
+        if (stopIfAnyPassed && passedAttemptsCount > 0) {
+          // we have a passing attempt and we should stop retrying and fail the test
+          result.outerTestStatus = 'failed'
+          result.forceState = currentAttemptResult === 'passed' ? currentAttemptResult : undefined
+          result.reasonToStop = 'FAILED_STOPPED_ON_FLAKE'
+        } else if (remainingAttempts === 0) {
+          // we have no remaining attempts and we should stop retrying and fail the test
+          result.outerTestStatus = 'failed'
+          result.forceState = currentAttemptResult === 'passed' ? currentAttemptResult : undefined
+          result.reasonToStop = 'FAILED_REACHED_MAX_RETRIES'
+        } else {
+          // we have remaining attempts and we should retry the test
+          result.final = false
+        }
+
+        break
+      default:
+        result.outerTestStatus = 'failed'
+        result.forceState = currentAttemptResult === 'passed' ? currentAttemptResult : undefined
```

Comment:
if the `reasonToStop` here is `FAILED_NO_RETRIES` then  `currentAttemptResult` will never be `passed` so will this line always be `undefined`?